spec/amp-cors-requests.md

```diff
@@ -37,7 +37,9 @@ by default.
 
 ## CORS Security in AMP
 
-This security protocol consists of two components: CORS origin and source origin restrictions.
+This security protocol consists of three components: `AMP-Same-Origin`, CORS origin and source origin restrictions.
+
+On same origin requests, AMP will set `AMP-Same-Origin: true` custom header. If this header is set it indicates the request is coming from same origin.
```

Comment:
Given the thread below - we need to check (and document checking) `Origin` header first and only when it's not available to look for `AMP-Same-Origin`.